src/utils/useEvents.ts

```diff
@@ -17,9 +18,13 @@ const useEvents = <T extends AllLeafletInstances | Map>(
   instance: T | null,
   events?: LeafletEventHandlerFnMap,
 ) => {
+  const mapInstance = useMapInstance()
+
+  // The eventsArray should only be set once
+  const eventsArray = useMemo(() => Object.entries(events || {}), [])
+
   useEffect(() => {
-    const eventsArray = Object.entries(events || {})
-    if (eventsArray.length) {
+    if (mapInstance && eventsArray.length) {
```

Comment:
The `length` check can be removed as you can call `forEach` on an empty array.

```suggestion
    if (mapInstance) {
```